internal/services/hpccache/hpc_cache_blob_nfs_target_resource.go

```diff
@@ -221,7 +221,7 @@ func resourceHPCCacheBlobNFSTargetDelete(d *pluginsdk.ResourceData, meta interfa
 		return err
 	}
 
-	future, err := client.Delete(ctx, id.ResourceGroup, id.CacheName, id.Name)
+	future, err := client.Delete(ctx, id.ResourceGroup, id.CacheName, id.Name, "")
```

Comment:
The Delete method adds a new parameter `force`, when given an empty string, it will behave like old way, which will not add it in the request and delete the storage target only when cache is flushed.
[Sdk code](https://github.com/Azure/azure-sdk-for-go/blob/f3c695847ff5d4c8ee2b432015a2577dff7b1a53/services/storagecache/mgmt/2021-09-01/storagecache/storagetargets.go#L200)
[Doc](https://docs.microsoft.com/en-us/rest/api/storagecache/storage-targets/delete)